planning_interface/move_group_interface/src/move_group.cpp

```diff
@@ -161,26 +161,35 @@ class MoveGroup::MoveGroupImpl
     ros::Time start_time = ros::Time::now();
     while (start_time == ros::Time::now())
     {
-      ros::WallDuration(0.01).sleep();
+      ros::WallDuration(0.001).sleep();
       ros::spinOnce();
+      if( node_handle_.getCallbackQueue() ) {
+        ( ( ros::CallbackQueue * ) node_handle_.getCallbackQueue())->callAvailable();
+      }
```

Comment:
reading the source code of `ros::NodeHandle` [here](https://github.com/ros/ros_comm/blob/kinetic-devel/clients/roscpp/include/ros/node_handle.h#L184), it looks like we should be able to ***replace***
```
ros::spinOnce();
```
by your proposed
```
( ( ros::CallbackQueue * ) node_handle_.getCallbackQueue()))->callAvailable();
```
without the additional check. getCallbackQueue returns the global queue if the handle does not have a queue on its own.